Produlytics/ui/front-end/src/app/model/http-interceptor/http-interceptor.service.ts

```diff
@@ -32,18 +35,37 @@ export class XhrInterceptor implements HttpInterceptor {
     req = req.clone({
       headers: req.headers.set('X-Requested-With', 'XMLHttpRequest')
     });
-    return next.handle(req)
-    .pipe(
-      catchError(
-        (error: HttpErrorResponse) => {
-          if (error.status === 401) {
-              this.loginService.logout();
 
-              return of();
-          }
+    /**
+     * Spring Boot di default è impostato in modo tale da bloccare le richieste REST che fanno del side effect
+     * (POST, PUT, DELETE, PATCH). Il modo per attivarle è mettere un header che si chiami X-XSFR-TOKEN e valorizzarlo
+     * con un cookie che ci dà il server. Questo è quanto si evince dalla documentazione di Spring Boot e da questo
+     * articolo https://www.baeldung.com/spring-security-csrf#2-front-end-configuration, da cui ho preso questa
+     * soluzione.
+     *
+     * Tutto ciò è parzialmente vero perché all'inizio il server non bloccava le POST, ma solo le PUT. Quindi potrebbe
+     * esserci dietro un bug di Spring Boot che non le impedisce. Resta il fatto che mettendo
+     * queste due linee di codice ora tutto funziona.
+     *
+     * Il cookie non dovrebbe essere HttpOnly, quindi non dovrebbe dare problemi questa get.
+     * */
+    const csrfToken = this.cookieService.get('XSRF-TOKEN');
+    if (csrfToken) {
+      req.headers.set('X-XSRF-TOKEN', csrfToken);
+    }
+
+    return next.handle(req)
+      .pipe(
+        catchError(
+          (error: HttpErrorResponse) => {
+            // if (error.status === 401) {
+            //     this.loginService.logout();
+            //
+            //     return of();
+            // }
```

Comment:
Dovresti decidere come gestire questo errore. Se questo pezzo di codice rimane commentato allora non serve neanche il `pipe` + `catchError` + `throwError`.

Reply:
Le opzioni sono 2: 

- Tolgo il codice commentato + `pipe` & company
- Lascio tutto ma bisogna fare in modo che il back-end ritorni 403 invece che 401 per i vari errori che sorgono nella modifica password

Reply:
Se la decisione ricade solo su di me allora sarei per la prima opzione